tests/test_aggregates.py

```diff
@@ -268,3 +270,38 @@ def _select(job):
             if _select(job):
                 selected_jobs.append((job,))
         assert list(aggregate_instance) == selected_jobs
+
+    def test_storing_hashing(self, setUp, project, list_of_aggregates):
+        # Since we need to store groups on a per aggregate basis in the project,
+        # we need to be sure that the aggregates are hashing and compared correctly.
+        # This test ensures this feature.
+        def _create_storing(aggregator):
+            return aggregator._create_AggregatesStore(project)
+
+        list_of_storing = list(map(_create_storing, list_of_aggregates))
+        # The above list contains 9 distinct storing objects and some duplicates.
+        # When this list is converted to set, then these objects are hashed first and
+        # then compared. Since sets don't carry duplicate values, we test whether the
+        # length of the set obtained from the list is equal to 9 or not.
+        assert len(set(list_of_storing)) == 9
+
+    def test_get_by_id(self, setUp, project, list_of_aggregates):
+        def _create_storing(aggregator):
+            return aggregator._create_AggregatesStore(project)
+
+        list_of_storing = set(map(_create_storing, list_of_aggregates))
+        for stored_aggregate in list_of_storing:
+            for aggregate in stored_aggregate:
+                assert aggregate == stored_aggregate[get_aggregate_id(aggregate)]
+
+    def test_get_invalid_id(self, setUp, project):
+        aggregator_instance = aggregator()._create_AggregatesStore(project)
+        for job in project:
+            with pytest.raises(LookupError):
+                aggregator_instance[get_aggregate_id((job,))]
+
+    def test_contains(self, setUp, project):
+        aggregator_instance = aggregator()._create_AggregatesStore(project)
+        assert tuple(project) in aggregator_instance
+        for job in project:
+            assert not (job,) in aggregator_instance
```

Comment:
Only testing one or two of these would be fine. We should also test that the default aggregate store handles getitem and contains correctly as well.